lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -246,12 +252,20 @@ public TaskStatusProto ToProto()
                 }
                 else if (_lastException.IsPresent())
                 {
+                    byte[] error;
+                    try
+                    {
+                        error = SerializeException(_lastException.Value);
+                    }
+                    catch (SerializationException)
+                    {
+                        error = SerializeException(new NonSerializableTaskException(_lastException.Value.ToString()));
```

Comment:
The behavior is then consistent across both failure at serialization and deserialization.